src/storage/Backend.cc

```diff
@@ -46,6 +70,23 @@ void ResultCallback::Complete(OperationResult res) {
     trigger->Release();
 }
 
+void ResultCallback::Init(detail::OperationMetrics* m) {
+    operation_metrics = m;
+    start_time = util::current_time(true);
+}
+
+void ResultCallback::UpdateOperationMetrics(EnumValPtr c) {
+    if ( operation_metrics ) {
+        if ( c == ReturnCode::SUCCESS )
+            operation_metrics->success->Inc();
+        else if ( c != ReturnCode::IN_PROGRESS )
+            operation_metrics->fail->Inc();
```

Comment:
One more thought: A `ReturnCode::KEY_EXISTS` or `ReturnCode::KEY_NOT_FOUND` will be tracked in the fail metric, but they don't really seem to be failures. Maybe "result" could be "hit" and "miss" for `Get()` instead of just "success". For `Put()` - I'm missing good words. Maybe "store" or "accept" and "exist" or "drop".